lib/application.js

```diff
@@ -811,6 +811,16 @@ Box.Application = (function() {
 		//----------------------------------------------------------------------
 
 		/**
+		 * Overrides default error handler
+		 * @param {Function} exceptionHandler handling function that takes an
+		 * exception as argument.
+		 * @returns {void}
+		 */
+		setErrorHandler: function(exceptionHandler) {
+			error = exceptionHandler;
```

Comment:
Here's the approach I would take:

Create a variable at the top of this file, next to `initialized = false;`

var customErrorHandler;

then you could do:
`customErrorHandler = exceptionHandler;`

then do:
```
function error(exception) {
    if (customErrorHandler) {
         customErrorHandler(exception);
         return;
    }

    if (globalConfig.debug) ...
}
```

This will maintain the integrity of Box.Application while still giving you a handle/callback